api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -693,19 +697,27 @@ protected String paramToProp(@NotNull String param) {
 			return Optional.of(ilike(propertyName, param.getValue(), MatchMode.EXACT));
 		} else if (param.isContains()) {
 			return Optional.of(ilike(propertyName, param.getValue(), MatchMode.ANYWHERE));
-		} else {
-			return Optional.of(ilike(propertyName, param.getValue(), MatchMode.START));
 		}
+		
+		return Optional.of(ilike(propertyName, param.getValue(), MatchMode.START));
+	}
+	
+	protected Optional<Criterion> propertyLike(@NotNull String propertyName, TokenParam tag) {
+		if (tag == null) {
+			return Optional.empty();
+		}
+		
+		return propertyLike(propertyName, new StringParam(tag.getValue()));
```

Comment:
I don't think we should actually be adding this. For `TokenParam` unlike `StringParam`, the match should always be equality, i.e., so instead of `propertyLike("t.uuid", tag)`, we should really just be doing `eq("t.uuid", tag)`.